apps/backend/src/modules/rating/typedefs/rating.ts

```diff
@@ -0,0 +1,123 @@
+import { gql } from "graphql-tag";
+
+const typedef = gql`
+enum MetricName {
+    "Numerical metrics"
+    Usefulness
+    Difficulty
+    Workload
+
+    "Boolean metrics"
+    Attendance
+    Recording
+}
+
+enum Semester {
+    Fall
+    Spring
+    Summer
+    Winter
+}
+
+"""
+Ratings by class
+"""
+type AggregatedRatings {
+    "Class identifer"
+    subject: String!
+    courseNumber: String!
+    semester: Semester!
+    year: Int!
+    class: String!
+
+    metrics: [Metric!]!
+}
+type Metric {
+    metricName: MetricName!
+    count: Int!
+    weightedAverage: Float!
+    categories: [Category!]!
+}
+type Category {
+    value: Int!
+    count: Int!
+}
+
+"""
+Ratings by user
+"""
+type UserRatings {
+    createdBy: String!
+    count: Int!
+    classes: [UserClass!]!
+}
+type UserClass {
+    "Class Identifiers"
+    subject: String!
+    courseNumber: String! 
+    semester: Semester!
+    year: Int!
+    class: String!
+
+    metrics: [UserMetric!]!
+}
+type UserMetric {
+    metricName: MetricName!
+    value: Int!
+}
+type SemesterAvailable {
+    year: Int!
+    semester: Semester!
+}
+
+"""
+Get data
+"""
+input ClassIdentifier {
+    "Class Identifiers"
+    subject: String!
+    courseNumber: String!
+    semester: Semester!
+    year: Int!
+    class: String!
+}
+type Query {
+    aggregatedRatings(
+        classIdentifier: ClassIdentifier!
```

Comment:
I would suggest flattening this input and writing it as `aggregatedRatings(subject: String!, courseNumber: string!, ...)` so as to not make overly-nested queries.